src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -418,13 +428,21 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
                 $markingStoreDefinition = new Reference($workflow['marking_store']['service']);
             }
 
-            $workflowDefinition = new DefinitionDecorator('workflow.abstract');
+            $definitionDefinition->addTag('workflow.definition', array(
+                'name' => $name,
+                'type' => $type,
+                'marking_store' => isset($workflow['marking_store']['type']) ? $workflow['marking_store']['type'] : null,
+            ));
+            $definitionDefinition->setPublic(false);
```

Comment:
Oh that's right. Interesting. I was considering if it would be a feature to have it public and mutable. But it isnt, If you want to modify the workflow you should do it with guards. 

I think we should leave this private for now to be able to postpone a solution for later. I also think we should introduce a `DefinitionInterface` with: 

* getInitialPlace()
* getPlaces()
* getTransitions()

That would allow us to have an `ImmutableDefinition` class if we later decide that is the better solution.